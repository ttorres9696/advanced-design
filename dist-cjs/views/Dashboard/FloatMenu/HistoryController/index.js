"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _react = _interopRequireDefault(require("react"));

var _reactRedux = require("react-redux");

var _RedoButton = _interopRequireDefault(require("./RedoButton"));

var _ResetButton = _interopRequireDefault(require("./ResetButton"));

var _styles = _interopRequireDefault(require("./styles"));

var _UndoButton = _interopRequireDefault(require("./UndoButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HistoryController = function HistoryController() {
  var classes = (0, _styles.default)();
  var historyLength = (0, _reactRedux.useSelector)(function (state) {
    return state.stage.selectedModuleSpec ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec.series].history.length : 0;
  });
  var activeHistoryIndex = (0, _reactRedux.useSelector)(function (state) {
    return state.stage.selectedModuleSpec ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec.series].activeHistoryIndex : -1;
  });
  return /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: classes.controller
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: classes.buttonGroup
  }, /*#__PURE__*/_react.default.createElement(_UndoButton.default, {
    activeHistoryIndex: activeHistoryIndex
  }), /*#__PURE__*/_react.default.createElement(_RedoButton.default, {
    historyLength: historyLength,
    activeHistoryIndex: activeHistoryIndex
  })), /*#__PURE__*/_react.default.createElement(_ResetButton.default, {
    activeHistoryIndex: activeHistoryIndex
  }));
};

var _default = /*#__PURE__*/_react.default.memo(HistoryController);

exports.default = _default;