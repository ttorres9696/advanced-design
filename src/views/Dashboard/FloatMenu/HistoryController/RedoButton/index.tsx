import clsx from 'clsx';
import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { redoHistory } from '../../../../../redux/stage/actions';
import useStyles from './styles';
import { RedoButtonProps } from './types';

const RedoButton: React.FC<RedoButtonProps> = ({ historyLength, activeHistoryIndex }) => {
  const classes = useStyles();

  const dispatch = useDispatch();

  const handleOnClick = useCallback(() => dispatch(redoHistory()), [dispatch]);

  const disabled = activeHistoryIndex >= historyLength - 1;

  return (
    <button className={classes.button} onClick={handleOnClick} disabled={disabled}>
      <i
        className={clsx(classes.icon, {
          [classes.disabledIcon]: disabled,
        })}
      />
      REDO
    </button>
  );
};

export default React.memo(RedoButton);
