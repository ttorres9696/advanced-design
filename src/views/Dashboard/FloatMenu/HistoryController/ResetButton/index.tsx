import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { resetHistory } from '../../../../../redux/stage/actions';
import useStyles from './styles';
import { ResetButtonProps } from './types';

const ResetButton: React.FC<ResetButtonProps> = ({ activeHistoryIndex }) => {
  const classes = useStyles();

  const dispatch = useDispatch();

  const handleOnClick = useCallback(() => dispatch(resetHistory()), [dispatch]);

  return (
    <button className={classes.button} onClick={handleOnClick} disabled={activeHistoryIndex < 0}>
      RESET
    </button>
  );
};

export default React.memo(ResetButton);
