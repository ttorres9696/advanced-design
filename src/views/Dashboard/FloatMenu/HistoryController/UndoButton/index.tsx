import clsx from 'clsx';
import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { undoHistory } from '../../../../../redux/stage/actions';
import undoIcon from '../../../../../shared/assets/undo_icon.svg';
import useStyles from './styles';
import { UndoButtonProps } from './types';

const UndoButton: React.FC<UndoButtonProps> = ({ activeHistoryIndex }) => {
  const classes = useStyles();

  const dispatch = useDispatch();

  const handleOnClick = useCallback(() => dispatch(undoHistory()), [dispatch]);

  const disabled = activeHistoryIndex < 0;

  return (
    <button className={classes.button} onClick={handleOnClick} disabled={disabled}>
      <i
        className={clsx(classes.icon, {
          [classes.disabledIcon]: disabled,
        })}
      />
      UNDO
    </button>
  );
};

export default React.memo(UndoButton);
