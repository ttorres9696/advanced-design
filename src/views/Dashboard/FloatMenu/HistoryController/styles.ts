import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    controller: {
      display: 'flex',
      flexDirection: 'column',
    },
    buttonGroup: {
      display: 'flex',
      flexDirection: 'row',
      marginBottom: 1,
    },
  }),
);
