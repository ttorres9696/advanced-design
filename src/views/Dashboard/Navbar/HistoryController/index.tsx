import { Box, IconButton } from '@material-ui/core';
import { Redo as RedoIcon, Undo as UndoIcon } from '@material-ui/icons';
import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from '../../../../redux';
import { redoHistory, undoHistory } from '../../../../redux/stage/actions';
import useStyles from './styles';

const HistoryController: React.FC = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const historyLength = useSelector((state: RootState) =>
    state.stage.selectedModuleSpec
      ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec!.series].history.length
      : 0,
  );
  const activeHistoryIndex = useSelector((state: RootState) =>
    state.stage.selectedModuleSpec
      ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec!.series].activeHistoryIndex
      : -1,
  );

  const onUndoHistoryClick = useCallback(() => dispatch(undoHistory()), [dispatch]);
  const onRedoHistoryClick = useCallback(() => dispatch(redoHistory()), [dispatch]);

  return (
    <Box className={classes.controller}>
      <IconButton
        color="inherit"
        aria-label="undo-history"
        onClick={onUndoHistoryClick}
        disabled={activeHistoryIndex < 0}
      >
        <UndoIcon />
      </IconButton>

      <IconButton
        color="inherit"
        aria-label="redo-history"
        onClick={onRedoHistoryClick}
        disabled={activeHistoryIndex >= historyLength - 1}
      >
        <RedoIcon />
      </IconButton>
    </Box>
  );
};

export default React.memo(HistoryController);
