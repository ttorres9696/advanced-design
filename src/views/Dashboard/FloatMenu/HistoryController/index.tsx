import { Box } from '@material-ui/core';
import React from 'react';
import { useSelector } from 'react-redux';

import { RootState } from '../../../../redux';
import RedoButton from './RedoButton';
import ResetButton from './ResetButton';
import useStyles from './styles';
import UndoButton from './UndoButton';

const HistoryController: React.FC = () => {
  const classes = useStyles();

  const historyLength = useSelector((state: RootState) =>
    state.stage.selectedModuleSpec
      ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec!.series].history.length
      : 0,
  );
  const activeHistoryIndex = useSelector((state: RootState) =>
    state.stage.selectedModuleSpec
      ? state.stage.moduleSpecsStages[state.stage.selectedModuleSpec!.series].activeHistoryIndex
      : -1,
  );

  return (
    <Box className={classes.controller}>
      <Box className={classes.buttonGroup}>
        <UndoButton activeHistoryIndex={activeHistoryIndex} />
        <RedoButton historyLength={historyLength} activeHistoryIndex={activeHistoryIndex} />
      </Box>

      <ResetButton activeHistoryIndex={activeHistoryIndex} />
    </Box>
  );
};

export default React.memo(HistoryController);
