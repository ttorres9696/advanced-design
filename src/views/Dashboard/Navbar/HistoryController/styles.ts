import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    controller: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
    },
  }),
);
