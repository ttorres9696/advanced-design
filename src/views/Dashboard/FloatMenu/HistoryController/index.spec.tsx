import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Provider } from 'react-redux';

import mockStore from '../../../../redux/specs/mocks/state.mock';
import HistoryController from '.';

let wrapper: ReactWrapper;

beforeEach(() => {
  wrapper = mount(
    <Provider store={mockStore}>
      <HistoryController />
    </Provider>,
    {},
  );
});

describe('HistoryController', () => {
  it('renders correctly', () => expect(toJson(wrapper)).toMatchSnapshot());
});
