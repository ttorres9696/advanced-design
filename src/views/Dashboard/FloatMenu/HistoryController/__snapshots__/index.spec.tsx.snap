// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryController renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(HistoryController)>
    <Styled(MuiBox)
      className="makeStyles-controller-1"
    >
      <div
        className="MuiBox-root MuiBox-root-3 makeStyles-controller-1"
      >
        <WithStyles(ForwardRef(IconButton))
          aria-label="undo-history"
          color="inherit"
          disabled={true}
          onClick={[Function]}
        >
          <ForwardRef(IconButton)
            aria-label="undo-history"
            classes={
              Object {
                "colorInherit": "MuiIconButton-colorInherit",
                "colorPrimary": "MuiIconButton-colorPrimary",
                "colorSecondary": "MuiIconButton-colorSecondary",
                "disabled": "Mui-disabled",
                "edgeEnd": "MuiIconButton-edgeEnd",
                "edgeStart": "MuiIconButton-edgeStart",
                "label": "MuiIconButton-label",
                "root": "MuiIconButton-root",
                "sizeSmall": "MuiIconButton-sizeSmall",
              }
            }
            color="inherit"
            disabled={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ButtonBase))
              aria-label="undo-history"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-colorInherit Mui-disabled"
              disabled={true}
              focusRipple={true}
              onClick={[Function]}
            >
              <ForwardRef(ButtonBase)
                aria-label="undo-history"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit Mui-disabled"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                disabled={true}
                focusRipple={true}
                onClick={[Function]}
              >
                <button
                  aria-label="undo-history"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <ForwardRef>
                      <WithStyles(ForwardRef(SvgIcon))>
                        <ForwardRef(SvgIcon)
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction",
                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                              "colorError": "MuiSvgIcon-colorError",
                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                              "root": "MuiSvgIcon-root",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
                            />
                          </svg>
                        </ForwardRef(SvgIcon)>
                      </WithStyles(ForwardRef(SvgIcon))>
                    </ForwardRef>
                  </span>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(IconButton)>
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="redo-history"
          color="inherit"
          disabled={true}
          onClick={[Function]}
        >
          <ForwardRef(IconButton)
            aria-label="redo-history"
            classes={
              Object {
                "colorInherit": "MuiIconButton-colorInherit",
                "colorPrimary": "MuiIconButton-colorPrimary",
                "colorSecondary": "MuiIconButton-colorSecondary",
                "disabled": "Mui-disabled",
                "edgeEnd": "MuiIconButton-edgeEnd",
                "edgeStart": "MuiIconButton-edgeStart",
                "label": "MuiIconButton-label",
                "root": "MuiIconButton-root",
                "sizeSmall": "MuiIconButton-sizeSmall",
              }
            }
            color="inherit"
            disabled={true}
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(ButtonBase))
              aria-label="redo-history"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-colorInherit Mui-disabled"
              disabled={true}
              focusRipple={true}
              onClick={[Function]}
            >
              <ForwardRef(ButtonBase)
                aria-label="redo-history"
                centerRipple={true}
                className="MuiIconButton-root MuiIconButton-colorInherit Mui-disabled"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                disabled={true}
                focusRipple={true}
                onClick={[Function]}
              >
                <button
                  aria-label="redo-history"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <ForwardRef>
                      <WithStyles(ForwardRef(SvgIcon))>
                        <ForwardRef(SvgIcon)
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction",
                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                              "colorError": "MuiSvgIcon-colorError",
                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                              "root": "MuiSvgIcon-root",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"
                            />
                          </svg>
                        </ForwardRef(SvgIcon)>
                      </WithStyles(ForwardRef(SvgIcon))>
                    </ForwardRef>
                  </span>
                </button>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(IconButton)>
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </Styled(MuiBox)>
  </Memo(HistoryController)>
</Provider>
`;
